indico/modules/events/timetable/controllers/manage.py

```diff
@@ -284,6 +328,38 @@ def _get_references(self, data: list[dict]) -> list[ContributionReference]:
                                                     value=entry['value']))
         return references
 
+    def _change_parent(self, session_block_id):
+        entry = self.contrib.timetable_entry
+        current_block = self.contrib.session_block
+
+        if session_block_id is None:
+            if current_block is None:
+                return
+            self.contrib.session_block = None
+            self.contrib.session = None
+            update_timetable_entry(entry, {
+                'parent': None,
+                'start_dt': entry.start_dt
+            })
+
+        if current_block is not None and current_block.id == session_block_id:
+            return
+
+        if session_block_id is not None:
+            target_block = self.event.get_session_block(session_block_id)
+            if target_block is None:
+                raise BadRequest('Session block does not exist')
+            if getattr(target_block.session, 'event_id', None) != self.event.id:
+                raise BadRequest('Session block does not belong to this event')
```

Comment:
```suggestion
```

This cannot happen because `get_session_block` only returns blocks that belong to the event.